### PR TITLE
add user settings page

### LINKED ISSUE #10: 10 - User Settings
As a user, I should be able to view my profile information and credentials and I should be able to edit these fields which include:

-     Name
-     Username
-     Password
-     E-Mail

### PR DESCRIPTION
**Pull request by spanjshiri**
This pr adds the user settings page which allows them to view and edit their username as well as editing their password. As more information gets associated to a user, this page will get updated.

This also adds the username displayed in the top right of the navbar when logged in.

This relates to issue #10 

Validation is added with the formik package in the frontend, but there is backend validation added as well.

### CODE REVIEW

**Review by mchughmk — DISMISSED** [review]

File: api/src/main/java/com/exelaration/abstractmemery/security/WebSecurity.java [hunk 1]
```diff
@@ -51,6 +52,8 @@ protected void configure(HttpSecurity http) throws Exception {
         .authorizeRequests()
         .antMatchers(HttpMethod.POST, SIGN_UP_URL)
         .permitAll()
+        .antMatchers(HttpMethod.PUT, SETTINGS_URL)
```
Comment (KatherynRizek) [inline api/src/main/java/com/exelaration/abstractmemery/security/WebSecurity.java:57]: Why was this added to the antMatchers? Shouldn't the user be logged in if they are looking at their settings?

Comment (spanjshiri) [reply api/src/main/java/com/exelaration/abstractmemery/security/WebSecurity.java:57]: Pretty sure I forgot to take this out when I was using it to debug. Nice catch I'll remove it now

**Review by KatherynRizek — CHANGES_REQUESTED** [review]
When I was testing locally, it gave me an error if I changed the username first and then went to change the password. Pretty sure the error was due to text remaining in the textbox from the username change since when I removed it, I could update the password. If there's a way to remove the text after the change, that would be best, but at least giving a helpful error message is needed

**Review by mchughmk — APPROVED** [review]

**Review by KatherynRizek — APPROVED** [review]
